internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -272,14 +271,16 @@ func (m ConfigurationResource) Read() sdk.ResourceFunc {
 				if files := prop.ProtectedFiles; files != nil {
 					configs := []ProtectedFile{}
 					for _, file := range *files {
-						if pointer.From(file.Content) != "" {
-							configs = append(configs, ProtectedFile{
-								Content:     pointer.ToString(file.Content),
-								VirtualPath: pointer.ToString(file.VirtualPath),
-							})
+						for _, protectedFile := range output.ProtectedFile {
+							if protectedFile.VirtualPath == pointer.ToString(file.VirtualPath) {
+								configs = append(configs, ProtectedFile{
+									Content:     protectedFile.Content,
+									VirtualPath: pointer.ToString(file.VirtualPath),
+								})
+								break
```

Comment:
The schema allows users to specify multiple protected files, I don't know whether this is a valid configuration from the perspective of the API but if it is and we're breaking out of the for loop here, then this will break for users that have more than one protected file specified in their config

Reply:
Based on an offline discussion:

There was a slight misunderstanding here which led to the comment. Another review of it revealed an issue with drift detection and @stephybun and I talked over a possible fix for it which has been pushed.